src/main/java/me/jellysquid/mods/sodium/client/render/chunk/vertex/format/ChunkVertexType.java

```diff
@@ -3,7 +3,11 @@
 import me.jellysquid.mods.sodium.client.gl.attribute.GlVertexFormat;
 
 public interface ChunkVertexType {
-    GlVertexFormat<ChunkMeshAttribute> getVertexFormat();
+    GlVertexFormat<?> getVertexFormat();
 
     ChunkVertexEncoder getEncoder();
+
+    default String getDefine() {
+        return "VERTEX_FORMAT_COMPACT"; // as in original Sodium all vertex formats would use this mode
```

Comment:
what is this comment saying?

Reply:
We need to allow vertex formats to specify a define so that the shader logic can be adjusted accordingly. However, we must also provide a default implementation, as otherwise existing mods (~~cough Iris, if I recall correctly~~) will crash with an abstract method error. We choose `VERTEX_FORMAT_COMPACT`, as any mods that were designed for an older version of Sodium expect that Sodium's shaders will be reading the vertex data in compact format.

Reply:
That makes sense but it's not really clear from the comment as it stands.